### PR TITLE
[DO-NOT-MERGE] Exceptions without error classes in SQL golden files

### PR DESCRIPTION
**Pull request by MaxGekk**

### CODE REVIEW

File: sql/core/src/test/resources/sql-tests/results/extract.sql.out [hunk 1]
```diff
@@ -317,7 +317,12 @@ select extract(not_supported from c) from t
 struct<>
 -- !query output
 org.apache.spark.sql.AnalysisException
-Literals of type 'not_supported' are currently not supported for the string type.; line 1 pos 7
+{
+  "errorClass" : "LEGACY",
+  "messageParameters" : {
```
Comment (MaxGekk) [inline sql/core/src/test/resources/sql-tests/results/extract.sql.out:322]: SPARK-41182

File: sql/core/src/test/resources/sql-tests/results/extract.sql.out [hunk 2]
```diff
@@ -326,7 +331,12 @@ select extract(not_supported from i) from t
 struct<>
 -- !query output
 org.apache.spark.sql.AnalysisException
-Literals of type 'not_supported' are currently not supported for the interval year to month type.; line 1 pos 7
+{
+  "errorClass" : "LEGACY",
+  "messageParameters" : {
```
Comment (MaxGekk) [inline sql/core/src/test/resources/sql-tests/results/extract.sql.out:336]: SPARK-41182

File: sql/core/src/test/resources/sql-tests/results/extract.sql.out [hunk 3]
```diff
@@ -335,7 +345,12 @@ select extract(not_supported from j) from t
 struct<>
 -- !query output
 org.apache.spark.sql.AnalysisException
-Literals of type 'not_supported' are currently not supported for the interval day to second type.; line 1 pos 7
+{
+  "errorClass" : "LEGACY",
+  "messageParameters" : {
```
Comment (MaxGekk) [inline sql/core/src/test/resources/sql-tests/results/extract.sql.out:350]: SPARK-41182

File: sql/core/src/test/resources/sql-tests/results/extract.sql.out [hunk 4]
```diff
@@ -924,7 +939,12 @@ select extract(DAY from interval '2-1' YEAR TO MONTH)
 struct<>
 -- !query output
 org.apache.spark.sql.AnalysisException
-Literals of type 'DAY' are currently not supported for the interval year to month type.; line 1 pos 7
+{
+  "errorClass" : "LEGACY",
+  "messageParameters" : {
+    "message" : "Literals of type 'DAY' are currently not supported for the interval year to month type.; line 1 pos 7"
```
Comment (MaxGekk) [inline sql/core/src/test/resources/sql-tests/results/extract.sql.out:945]: SPARK-41182

**Comment by MaxGekk** [comment]
@panbingkun @LuciferYang @itholic @cloud-fan @srielau @anchovYu @entong I changed the mechanism of generating SQL golden files in this PR to detect the exceptions that haven't been ported on error classes. There are still a few exceptions for which I opened the JIRAs:
- SPARK-41172
- SPARK-41173
- SPARK-41174
- SPARK-41175
- SPARK-41176
- SPARK-41179
- SPARK-41180
- SPARK-41181
- SPARK-41182

**Comment by panbingkun** [comment]
OK

**Comment by LuciferYang** [comment]
- SPARK-41173: https://github.com/apache/spark/pull/38705

File: sql/core/src/test/resources/sql-tests/results/udaf.sql.out [hunk 5]
```diff
@@ -48,7 +53,12 @@ SELECT default.udaf1(int_col1) as udaf1 from t1
 struct<>
 -- !query output
 org.apache.spark.sql.AnalysisException
-Can not load class 'test.non.existent.udaf' when registering the function 'spark_catalog.default.udaf1', please make sure it is on the classpath; line 1 pos 7
+{
+  "errorClass" : "LEGACY",
+  "messageParameters" : {
+    "message" : "Can not load class 'test.non.existent.udaf' when registering the function 'spark_catalog.default.udaf1', please make sure it is on the classpath; line 1 pos 7"
```
[the remark this answers is not in the document]
Comment (MaxGekk) [reply sql/core/src/test/resources/sql-tests/results/udaf.sql.out:59]: @panbingkun Any ideas why we still don't see the error class `CANNOT_LOAD_FUNCTION_CLASS ` introduced by your PR https://github.com/apache/spark/pull/38696?

**Comment by MaxGekk** [comment]
> @MaxGekk Have all the issues mentioned in this pr been solved?

Let me rebase it on the recent master and regenerate golden files, then we will see.

File: sql/core/src/test/resources/sql-tests/results/udaf.sql.out [hunk 6]
```diff
@@ -31,7 +31,12 @@ SELECT default.myDoubleAvg(int_col1, 3) as my_avg from t1
 struct<>
 -- !query output
 org.apache.spark.sql.AnalysisException
-Invalid number of arguments for function spark_catalog.default.mydoubleavg. Expected: 1; Found: 2; line 1 pos 7
+{
+  "errorClass" : "LEGACY",
+  "messageParameters" : {
+    "message" : "Invalid number of arguments for function spark_catalog.default.mydoubleavg. Expected: 1; Found: 2; line 1 pos 7"
```
[the remark this answers is not in the document]
Comment (MaxGekk) [reply sql/core/src/test/resources/sql-tests/results/udaf.sql.out:37]: This one should disappear after https://github.com/apache/spark/pull/38919

**Comment by MaxGekk** [comment]
All issues have been fixed. At the moment, all sql tests (in *.sql files) raise exception with error classes. I would propose to merge this change to detect exceptions that are not ported on error classes yet. @cloud-fan Are you ok with that?

**Pull request closed without merging**